ui/components/failed.jsx

```diff
@@ -0,0 +1,175 @@
+  Jumbotron, Grid, Row, Col, Panel}
+  from 'react-bootstrap';
+
+function _renderDefaultView(paths) {
+  return <div className="simple">
+    <img className="diff"
+         src={paths.diff}
+         key={paths.diff} />
+  </div>;
+}
+
+function _render2UpView(paths) {
+  return <div className="simple">
+    <Grid>
+      <Row className="show-grid">
+        <Col xs={12} md={6}>
+          <img className="baseline"
+               src={paths.baseline}
+               key={paths.baseline} />
+        </Col>
+        <Col xs={12} md={6}>
+          <img className="screenshot"
+               src={paths.screenshot}
+               key={paths.screenshot} />
+        </Col>
+      </Row>
+    </Grid>
+  </div>;
+}
+
+function _renderSwipeView(paths, value, onValueChange) {
+  return <div className="special">
+    <ImageDiff before={paths.screenshot}
+               after={paths.baseline}
+               type="swipe"
+               value={value} />
+    <Clearfix />
+    <input type="range"
+           min={0}
+           max={1}
+           step={.01}
+           defaultValue={value}
+           onChange={onValueChange} />
+  </div>;
+}
+
+function _renderFadeView(paths, value, onValueChange) {
+  return <div className="special">
+    <ImageDiff before={paths.screenshot}
+               after={paths.baseline}
+               type="fade"
+               value={value} />
+    <Clearfix />
+    <input type="range"
+           min={0}
+           max={1}
+           step={.01}
+           defaultValue={value}
+           onChange={onValueChange} />
+  </div>;
+}
+
+function _getSelectViewButtons(viewOptions, currentView, onViewChange) {
+  let buttons = [];
+
+  viewOptions.forEach(function(item) {
+    buttons.push(
+      <Button name={item}
+              key={item}
+              onClick={onViewChange}
+              className={item === currentView ? 'active' : null}>
+        {item}
+      </Button>
+    );
+  });
+
+  return buttons;
+}
+
+class FailedTest extends React.Component {
+
+  render() {
+    const {paths, error} = this.props,
+        view = this.state.view;
+    let buttons = [],
+        report;
+
+    if (paths !== undefined) {
+
+      /* buttons to select the report type */
+      buttons = _getSelectViewButtons(this.statics.viewOptions,
+          view, this.onViewChange);
+
+      /* switch betwen report types */
+      report = this.statics.viewHandlers[view](paths,
+          this.state.value, this.onValueChange);
+    }
+
+    return <div className="diffs">
+      <Button bsStyle="danger"
+              bsSize="xsmall"
+              onClick={this.onErrorMessageOpen}>
+        Show Error
+      </Button>
+
+      <Panel collapsible expanded={this.state.openError} bsStyle="danger">
+        {paths === undefined
+          ? <p> This test did not fail because Mugshot found differences :( </p>
+          : <p> {error.name} : {error.message} </p> }
+      </Panel>
+
+      {paths !== undefined
```

Comment:
```js
{!_.isUndefined(paths) ? this.renderJumbotron() : null}
```

If i didn't miss something this will also get rid of the if at the beginning of this render method where you define `buttons` and `report` that are only used here. You can build those variables inside this new private method and remove them from `render()`